tests/providers/microsoft/azure/log/test_wasb_task_handler.py

```diff
@@ -98,7 +98,7 @@ def test_wasb_log_exists(self, mock_hook):
 
     @mock.patch("airflow.providers.microsoft.azure.hooks.wasb.WasbHook")
     def test_wasb_read(self, mock_hook):
-        mock_hook.return_value.read_file.return_value = 'Log line'
+        mock_hook.return_value.read_file.return_value = b'Log line'
```

Comment:
I'm not sure I understand the problem you're trying to solve. However, in this test, it's returning bytes, is that what you expect?
It'll help if you'd say the Azure provider version you saw this issue because we recently resolved an issue around this. Steps to replicate the issue will be very much appreciated.

Reply:
Hi the problem i'm solving is the the azure storage provider returns a utf encoded byte array and not a string. When storing logs in the remote storage you will get an error about converting bytes. 

Repo steps, configure a AzureStorage storage as a remote log source for airflow and this error will occur. The pods will fail but the task will be marked as success.

```

 File “/usr/local/lib/python3.8/logging/__init__.py”, line 2123, in shutdown
    h.close()
  File “/root/.local/lib/python3.8/site-packages/airflow/providers/microsoft/azure/log/wasb_task_handler.py”, line 97, in close
    self.wasb_write(log, remote_loc, append=True)
  File “/root/.local/lib/python3.8/site-packages/airflow/providers/microsoft/azure/log/wasb_task_handler.py”, line 178, in wasb_write
    log = ‘\n’.join([old_log, log]) if old_log else log
TypeError: sequence item 0: expected str instance, bytes found
```

If you look at their SDK all versions of their provider from 12 forward have this return type.

Reply:
Look at their docs: https://docs.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.blobclient?view=azure-python#download-blob-offset-none--length-none----kwargs- it shows the return type

Reply:
Here is the method it is using
`content_as_text
`
Here is the docs

content_as_text
Download the contents of this blob, and decode as text.

This operation is blocking until all data is downloaded.

Python

Copy
content_as_text(max_concurrency=1, encoding='UTF-8')

Reply:
![Screenshot 2021-03-26 at 12 02 08](https://user-images.githubusercontent.com/739819/112622353-3773b580-8e2b-11eb-84fe-3053ff16acd6.png)

Reply:
I was not able to reproduce this

Reply:
Interesting. As you can see from my logs and microsoft documentation, this is correct and what is there should not work. I have this fix running in my local instance of airflow as well and it works perfectly. I'm running python 3.8.

Reply:
What version of python and provider version are you using?

Reply:
Did you test with actually sending logs to blob storage or where you using a file share? I can't imagine that you could not reproduce.

Reply:
When we returned bytes, users complained of logs coming as single line, see https://github.com/apache/airflow/issues/14312. I'm wondering whether your issue has to do with something else? and you haven't said the airflow version you saw this issue

Reply:
>Notice in the screenshot it shows that it is bytes that are downloaded

From the screenshot String was returned

Reply:
> What version of python and provider version are you using?

Python 3.7 and master

Reply:
> What version of python and provider version are you using?

Python 3.7 and master

Reply:
> Did you test with actually sending logs to blob storage or where you using a file share? I can't imagine that you could not reproduce.

Blob storage

Reply:
Ok i'm working on setting up a repro for this. I'm using airflow 2.01. I'll get back to you

Reply:
I tried to set up a simple example of this scenario and I could not repro this. The only way this happens is on my running instance of airflow. Here is how I build the image

FROM apache/airflow:2.0.1-python3.8
COPY ./podtemplates /opt/airflow/templates

USER airflow
RUN pip install --user Flask-OpenID && \
    pip install --user authlib && \
    pip install --user cryptography && \
    pip install --user apache-airflow[crypto] && \
    pip install --user apache-airflow['wasb'] && \
    pip install --user azure-storage-blob===12.7.1 && \
    pip install --user lib && \
    pip install --user apache-airflow['kubernetes'] 

I have tried all verions of the blob client + 12

I can close this PR unless you can think of something to try. My workaround functions and we are running. It is just ugly that I have to overwrite the code to get this working like this

COPY hacks/wasb_task_handler.py /home/airflow/.local/lib/python3.8/site-packages/airflow/providers/microsoft/azure/log

Reply:
You don't need to install azure-storage-blob by yourself.
To install wasb on your airflow instance, just run `pip install apache-airflow-providers-microsoft-azure` on the instance. The latest version will be installed on the instance

Reply:
You can close the PR as I don't have anything else to try. We can always open it if we are able to reproduce it

Reply:
I tried to install that package initially and I received the error the blob storage provider didn't exist so I added it. I can't reproduce out side either so lets close it